src/main/scala/mesosphere/chaos/http/impl/HttpServiceImpl.scala

```diff
@@ -0,0 +1,74 @@
+package mesosphere.chaos.http.impl
+
+import java.util
+import javax.servlet.{Filter, Servlet, DispatcherType}
+
+import com.sun.jersey.api.core.DefaultResourceConfig
+import com.sun.jersey.spi.container.servlet.ServletContainer
+import org.eclipse.jetty.servlet.{FilterHolder, ServletHolder}
+import org.slf4j.LoggerFactory
+import mesosphere.chaos.http.HttpService
+
+import scala.util.Try
+/**
+  * Wrapper for starting and stopping the HttpServer.
+  */
+class HttpServiceImpl(httpServer: HttpServer) extends HttpService {
+  val servletPath = "/*"
+
+  private[this] val log = LoggerFactory.getLogger(getClass.getName)
+  private[this] var started = false
+
+  def startUp() {
+    log.debug("Starting up HttpServer.")
+    try {
+      started = true
+      preStartUp()
```

Comment:
I agree - that's why my suggestion with initialized. You have to do this only once, where started would suggest it changes over start/stop cycles. 

> Am 16.02.2016 um 17:18 schrieb Alexander Weber <notifications@github.com>:
> 
> In src/main/scala/mesosphere/chaos/http/impl/HttpServiceImpl.scala <https://github.com/mesosphere/chaos/pull/54#discussion_r53033743>:
> 
> > +
> > +import scala.util.Try
> > +/**
> > +  * Wrapper for starting and stopping the HttpServer.
> > +  */
> > +class HttpServiceImpl(httpServer: HttpServer) extends HttpService {
> > +  val servletPath = "/*"
> > +
> > +  private[this] val log = LoggerFactory.getLogger(getClass.getName)
> > +  private[this] var started = false
> > +
> > +  def startUp() {
> > +    log.debug("Starting up HttpServer.")
> > +    try {
> > +      started = true
> > +      preStartUp()
> Got that, but I also have a question on the shutDown() function: there is no initialised = false commando in there. In my opinion that is correct, as the /* servlet container, containing all registered resources, is already added to the servlet handler by this point. Calling the preStartUp() function as it is right now, would cause an error of adding a servlet to the same path. Do you agree on that?
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/mesosphere/chaos/pull/54/files#r53033743>.
>